live.tsx

```diff
@@ -91,22 +104,25 @@ export function createLivePage<Data>(options: LivePageOptions<Data>) {
     },
   };
 
-  function LivePage(
-    props: PageProps<LivePageData & Data>,
-  ) {
+  function LivePage(props: PageProps<LivePageData<Data>>) {
     const manifest = userManifest;
     const { data } = props;
-    const { components } = data;
-    const renderComponents = components && components.length > 0
-      ? components.map(({ component, props }: any) => {
-        const Comp = manifest.islands[`./islands/${component}.tsx`]?.default ||
-          manifest.components![`./components/${component}.tsx`]?.default;
-        return <Comp {...props} />;
-      })
-      : defaultRender
-      ? defaultRender(props)
-      : <div>Page not found</div>;
-    const InspectVSCode = !isDenoDeploy &&
+    const { components, loaderData } = data;
+    const renderComponents =
+      components && components.length > 0 ? (
+        components.map(({ component, props }: any) => {
+          const Comp =
+            manifest.islands[`./islands/${component}.tsx`]?.default ||
+            manifest.components![`./components/${component}.tsx`]?.default;
+          return <Comp {...props} loaderData={loaderData} />;
```

Comment:
Should pass loaderData to components?

Reply:
Well, for now, yes. It's straightforward and clear.